Tools/clinic/clinic.py

```diff
@@ -1628,10 +1628,16 @@ def parse_clinic_block(self, dsl_name):
         def is_stop_line(line):
             # make sure to recognize stop line even if it
             # doesn't end with EOL (it could be the very end of the file)
-            if not line.startswith(stop_line):
-                return False
-            remainder = line[len(stop_line):]
-            return (not remainder) or remainder.isspace()
+            if line.startswith(stop_line):
+                remainder = line[len(stop_line):]
+                if remainder.isspace():
+                    return True
+                else:
+                    fail(f"Garbage after stop line: {remainder!r}")
+            # gh-92256: don't allow incorrectly formatted stop lines
+            elif line.lstrip().startswith(stop_line):
+                fail("Whitespace is not allowed before the stop line: {stop_line!r}")
+            return False
```

Comment:
I would rewrite it in more symmetric way:
```suggestion
            if line.startswith(stop_line):
                remainder = line[len(stop_line):]
                if remainder and not remainder.isspace():
                    fail(f"Garbage after stop line: {remainder!r}")
                return True
            else:
                # gh-92256: don't allow incorrectly formatted stop lines
                if line.lstrip().startswith(stop_line):
                    fail("Whitespace is not allowed before the stop line: {stop_line!r}")
                return False
```

Reply:
We could also drop the `else`, but I'm fine with this as well :)

Reply:
Thanks!

Reply:
Fixup in 650392841091e62be25aafab7e43a031af6cdab0